### PR TITLE
Transaction monitoring

### PR DESCRIPTION
**Pull request by maricalucian**

### CODE REVIEW

File: src/redux/wallets/selectors.ts [hunk 1]
```diff
@@ -48,8 +48,10 @@ export const getAccountTransactions = (
     );
     const transactions = getSelectedWallet(state).transactions;
     if (transactions) {
-        return Object.values(getSelectedWallet(state).transactions).filter(
-            tx => tx.address === account.address
+        return Object.values(transactions).filter(
+            tx =>
+                tx.address.toLowerCase() === account.address.toLowerCase() ||
+                tx.toAddress.toLowerCase() === account.address.toLowerCase()
```
Comment (georgeburduhos) [inline src/redux/wallets/selectors.ts]: blockchain verification should be added here to make sure the address is not on different blockchains

File: src/core/blockchain/ethereum/client.ts [hunk 2]
```diff
@@ -141,6 +150,61 @@ export class Client extends BlockchainGenericClient {
...
+                    type: TransactionType.TRANSFER,
+
+                    address: txInfo.from,
+                    publicKey: '', // TODO: get publicKey form vrs
+
+                    toAddress: txInfo.to,
+                    amount: txInfo.value,
+                    feeOptions: {
+                        gasPrice: txInfo.gasPrice,
+                        feeTotal: txInfo.gas
```
[the remark this answers is not in the document]
Comment (maricalucian) [reply src/core/blockchain/ethereum/client.ts]: done

File: src/redux/wallets/actions.ts [hunk 3]
```diff
@@ -336,6 +375,85 @@ export const getBalance = (
...
+                `${transaction.status}: Transaction of ${formattedAmount} from ${formatAddress(
+                    transaction.address,
+                    blockchain
+                )} to ${formatAddress(transaction.toAddress, blockchain)}`,
+                {
+                    type: NotificationType.TRANSACTION_UPDATE,
+                    data: {
+                        walletId: wallet.id,
+                        accountIndex: notificatonAccount.index,
+                        token: notificatonAccount.tokens.ETH,
```
[the remark this answers is not in the document]
Comment (maricalucian) [reply src/redux/wallets/actions.ts]: updated. was before adding support for zilliqa

File: src/address-monitor/index.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            token: await getDeviceToken(),
+            type: Platform.OS === 'android' ? 'fcm' : 'apn',
+            address
+        })
+    };
+
+    fetch(url + '/add', request);
+};
+
+export const removeAddress = async (blockchain: Blockchain, address: string[]) => {
```
Comment (krisboit) [inline src/address-monitor/index.ts]: i think this should be removed, as it was removed from cloud function

File: src/core/blockchain/ethereum/utils.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { TransactionStatus } from '../../wallet/types';
+
+export const getEthStatusByCode = (status): TransactionStatus => {
```
Comment (krisboit) [inline src/core/blockchain/ethereum/utils.ts]: this should be renamed to `getTransactionStatusByCode()` and moved in `transactions.ts`.

File: src/core/blockchain/types/client.ts [hunk 6]
```diff
@@ -1,14 +1,25 @@
...
-import { IFeeOptions } from './transaction';
+import { IFeeOptions, IBlockchainTransaction } from './transaction';
 import { IBlockchainNameService } from '.';
 
 export interface IBlockInfo {
     number: number;
     hash?: string;
 }
 
+export interface ITransactionReceipt {
```
Comment (krisboit) [inline src/core/blockchain/types/client.ts]: i didn't saw this interface being used, if it's used move it under ethereum folder as it's ethereum specific

File: src/core/blockchain/zilliqa/client.ts [hunk 7]
```diff
@@ -136,6 +141,52 @@ export class Client extends BlockchainGenericClient {
...
+            feeOptions: {
+                gasPrice: txData.gasPrice,
+                gasLimit: txData.gasLimit,
+                feeTotal: txData.receipt?.cumulative_gas
+            },
+            broadcatedOnBlock: txData.receipt?.epoch_num,
+            nonce: txData.nonce,
+            status: txStatus
+        };
+        return null;
```
Comment (krisboit) [inline src/core/blockchain/zilliqa/client.ts]: this is not needed.

File: src/address-monitor/index.ts (hunk 4 above)
Comment (maricalucian) [reply src/address-monitor/index.ts]: done

File: src/core/blockchain/types/client.ts (hunk 6 above)
Comment (maricalucian) [reply src/core/blockchain/types/client.ts]: removed it now

File: src/core/blockchain/zilliqa/client.ts (hunk 7 above)
Comment (maricalucian) [reply src/core/blockchain/zilliqa/client.ts]: refactored

**Pull request merged**